src/intermediate_config.rs

```diff
@@ -0,0 +1,305 @@
+        Ok(content) => content[0].to_owned()
+    };
+
+    let remote_machine = match &yaml["remoteMachine"] {
+        Yaml::Hash(remote_machine) => {
+            let host = match &remote_machine.get(&Yaml::String(String::from("host"))) {
+                Some(host) => match host {
+                    Yaml::String(host) => Some(host.to_string()),
+                    Yaml::Null => None,
+                    _ => return Err(String::from("remoteMachine.host must be a string"))
+                },
+                None => None
+            };
+
+            Some(IntermediateRemoteMachine {
+                host,
+            })
+        }
+        Yaml::Null | Yaml::BadValue => None,
+        ref something_else => return Err(format!("'remoteMachine' must be an object, but was {:#?}", something_else))
+    };
+
+    let compression = match &yaml["compression"] {
+        Yaml::Hash(compression) => {
+            let local = match compression.get(&Yaml::String(String::from("local"))).cloned() {
+                Some(local) => match local {
+                    Yaml::Integer(local) => if local >= 1 && local <= 9 {
+                        Some(local)
+                    } else {
+                        return Err(format!("'compression.local' must be a positive integer from 1 to 9, but was {:#?}", local));
+                    },
+                    Yaml::Null | Yaml::BadValue => None,
+                    ref something_else => return Err(format!("'compression.local' must be a positive integer from 1 to 9, but was {:#?}", something_else))
+                },
+                None => None
+            };
+
+            let remote = match compression.get(&Yaml::String(String::from("remote"))).cloned() {
+                Some(remote) => match remote {
+                    Yaml::Integer(remote) => if remote >= 1 && remote <= 9 {
+                        Some(remote)
+                    } else {
+                        return Err(format!("'compression.remote' must be a positive integer from 1 to 9, but was {:#?}", remote));
+                    },
+                    Yaml::Null | Yaml::BadValue => None,
+                    ref something_else => return Err(format!("'compression.remote' must be a positive integer from 1 to 9, but was {:#?}", something_else))
+                },
+                None => None
+            };
+
+            Some(IntermediateCompression {
+                local,
+                remote,
+            })
+        }
+        Yaml::Null | Yaml::BadValue => None,
+        _ => return Err(String::from("'compression' must be an object"))
+    };
+
+    Ok(IntermediateConfig {
+        remote_machine,
+        compression,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_config_from_str_all_fields_2_spaces_indent() {
+        let content = "
+remoteMachine:
+  host: computer1
+compression:
+  local: 5
+  remote: 2"
```

Comment:
All right, in the PR description you mention `user` and `port`, but both seem non-existant in the implementation. And I guess we don’t need them since SSH handles it for us.

I think we have not discussed config contents yet. I see it this way.

```yaml
# remote, like Git
remote:
  name: "SSH name"
sending:
  compression: N
  ignore:
    - "path"
    - "path"
receiving:
  compression: N
  ignore:
    - "path"
    - "path"
# compression declared this way overrides sending and receiving values
compression: N
# compression declared this way extends sending and receiving values
ignore:
  - "path"
  - "path"
```

Do not forget that `.mainframer.yml` can be placed both at `${HOME}` and at `${PROJECT}`.

* Corporate mainframe, multiple projects.
    * Home config — basically everything. Project config — additional ignore rules.
* Corporate and personal mainframe, multiple projects.
    * Home config — personal thingies. Project config — additional ignore rules plus different remote and probably compression levels.

Basically what Git does, but with sharing in VCS.

And yes, I think we should move away from a `.mainframer` directory since a single file simplifies things.

Reply:
From the description:

```
Note that `remoteMachine.user` and `remoteMachine.port` are not implemented yet (we continue to rely on `~/.ssh/config` read by `ssh` for now)
```

Also, I've filed #240 and #241 for user and port properties (there is description explaining why)

Reply:
>And yes, I think we should move away from a .mainframer directory since a single file simplifies things.

That we can't do easily, similarly to `.gitignore` and `.git/config` you want to check in ignore lists into version control, while not checking in the config

Reply:
>Do not forget that .mainframer.yml can be placed both at ${HOME} and at ${PROJECT}.

I know lol, that's why I want to merge it so badly to implement $HOME part

Reply:
Done